MASFoundation/Classes/MAS.h

```diff
@@ -337,6 +318,27 @@
 
 # pragma mark - Gateway Monitoring
 
+/**
+ *  Sets the gateway monitoring block defined by the GatewayMonitorStatusBlock type.
+ *  This block will be triggered when any change to the current monitoring status
+ *  takes place with a MASGatewayMonitoringStatus.
+ *
+ *  The gateway monitoring status enumerated values are:
+ *
+ *      MASGatewayMonitoringStatusNotReachable
+ *      MASGatewayMonitoringStatusReachableViaWWAN
+ *      MASGatewayMonitoringStatusReachableViaWiFi
+ *
+ *  This is optional and it can be set to nil at any time to stop receiving the notifications.
+ *
+ *  Reachability status update can also be received as NSNotification.  Subscribe notification key, MASGatewayMonitorStatusUpdateNotification, and notifications will be broadcasted whenever there is a change in network reachability status for the primary gateway on the plaload of NSNotification as NSDictionary format @{ (NSString)"host" : (NSNumber)enumValue}:
```

Comment:
*  Reachability status update can also be received as NSNotification.  Subscribe notification key, MASGatewayMonitorStatusUpdateNotification, and notifications will be broadcasted whenever there is a change in network reachability status for the primary gateway on the payload of NSNotification as NSDictionary format @{ (NSString)"host" : (NSNumber)enumValue}: